client/my-sites/backup/backup-date-picker/hooks.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import { Moment } from 'moment';
+import { useCallback } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { useLocalizedMoment } from 'calypso/components/localized-moment';
+import { useDateWithOffset, useFirstMatchingBackupAttempt } from '../hooks';
+
+export const useCanGoToDate = ( selectedDate: Moment, oldestDateAvailable?: Moment ) => {
+	const moment = useLocalizedMoment();
+	const today = useDateWithOffset( moment() );
+
+	return useCallback(
+		( desiredDate ) => {
```

Comment:
Could we add the `Moment` type to the `desiredDate` parameter and add `boolean` as the return type of the function? It's clear that the function signature should be `( desiredDate: Moment) => boolean` but TS can only infer that it's `( desiredDate: any) => any`.